libraries/SITL/SIM_Precland.cpp

```diff
@@ -112,17 +113,23 @@ void SIM_Precland::update(const Location &loc, const Vector3d &position)
         return;
     }
 
+    Location origin_center;
```

Comment:
you should move back this block after the distance_z as we don't need location at this point, thus it same a little calculus

Reply:
I have to say that I don't like the use of the word "origin" when we're talking about "home".  People very often confuse them and this poor name contributes to that.  It's an existing bad name though so not a blocker.

Reply:
we could probably rename everything here as vehicle_home/base.

Reply:
changed all origins to "precland", hope that is okay